finetune/adapter.py

```diff
@@ -143,8 +144,10 @@ def main(fabric: L.Fabric, devices: int, seed: int, config: Config, data: LitDat
     save_path = out_dir / "final" / "lit_model.pth"
     save_path.parent.mkdir(parents=True, exist_ok=True)
     save_adapter_checkpoint(fabric, model, save_path)
-    # Copy checkpoint files from original checkpoint dir
-    copy_config_files(checkpoint_dir, save_path.parent)
+    if fabric.global_rank == 0:
```

Comment:
No, because if it were not a shared filesystem, then there are these scenarios:

1. ddp -> Fabric.save saves a checkpoint on global rank 0 only anyway. Other nodes/filesystems wouldn't get a file.
2. fsdp sharded checkpoints -> Fabric.save will save on local rank 0. This means you have several directories and need to combine the checkpoint folders from each node into one in the end. Anything that is not sharded should only be saved into checkpoint folder on node 0, because it doesn't make sense to replicate it (we do it like this also with meta.pt in FSDP). https://github.com/Lightning-AI/pytorch-lightning/blob/b871f7a826db015b8b909cf86302c6c4cbf18ec1/src/lightning/fabric/strategies/fsdp.py#L496-L497